From 54a8baaa0599876d7ee547f8dc349aa58f70425b Mon Sep 17 00:00:00 2001
From: Paul Harrison <pharrison@mythtv.org>
Date: Thu, 23 Oct 2014 11:15:47 +0100
Subject: [PATCH] MythMusic: wait for all music and playlists to load in the
 media handler

This should fix an occasional crash when a CD is in the drive when starting the FE.
---
 mythplugins/mythmusic/mythmusic/main.cpp |   11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/mythplugins/mythmusic/mythmusic/main.cpp b/mythplugins/mythmusic/mythmusic/main.cpp
index ef775d7..c835ad1 100644
--- a/mythplugins/mythmusic/mythmusic/main.cpp
+++ b/mythplugins/mythmusic/mythmusic/main.cpp
@@ -507,13 +507,16 @@ static void handleCDMedia(MythMediaDevice *cd)
     if (!gMusicData->initialized)
         gMusicData->loadMusic();
 
-    // remove any existing CD tracks
-    if (gMusicData->all_music)
+    // wait for the music and playlists to load
+    while (!gMusicData->all_playlists->doneLoading() || !gMusicData->all_music->doneLoading())
     {
-        gMusicData->all_music->clearCDData();
-        gMusicData->all_playlists->getActive()->removeAllCDTracks();
+        qApp->processEvents();
+        usleep(50000);
     }
 
+    gMusicData->all_music->clearCDData();
+    gMusicData->all_playlists->getActive()->removeAllCDTracks();
+
     // find any new cd tracks
     CdDecoder *decoder = new CdDecoder("cda", NULL, NULL);
     decoder->setDevice(newDevice);
-- 
1.7.10.2

